optimization/sggp/sggp-algorithm-with-test.metta

```diff
@@ -0,0 +1,597 @@
+;;An implementation of the paper called "Avoiding the Bloat with 
+;;Stochastic Grammar-based Genetic Programming"
+
+!(register-module! ../../../metta-moses)
+;; !(import! &self metta-moses:optimization:sggp:sggp-algorithm)
+!(import! &self metta-moses:utilities:general-helpers)
+!(import! &self metta-moses:scoring:fitness)
+!(bind! &grammarSpace (new-space))
+!(bind! &stochasticGrammarSpace (new-space))
+!(bind! &alphabetSpace (new-space))
+!(bind! &trainingSpace (new-space))
+!(bind! rand (py-atom random.randint)) ;;pseudo random integers
+!(bind! random (py-atom random.random)) ;;pseudo random real numbers from 0 to 1
+!(bind! pow (py-atom pow))
+(= (initializeAlphabetSpace $alphabetSpace)
+    (add-reduct $alphabetSpace (superpose (start expr bop uop term)))
+)
+;;The below procedure will initialize the grammarSpace with a cfg
+;; Written in context free grammar it will contain the following representation
+;; <start> -> <expr>
+;; <expr>  -> <bop> <expr> <expr> |
+;;            <uop> <expr> |
+;;            <term>
+;; <bop>   -> AND | OR
+;; <uop>   -> NOT
+(= (initializeGrammarSpace $grammarSpace)
+    (add-reduct $grammarSpace (superpose (
+            (start expr)
+            (expr (bop expr expr))
+            (expr (uop expr))
+            (expr term)
+            (bop AND)
+            (bop OR)
+            (uop NOT)
+
+        )
+    )
+    )
+)
+;; This function initializes the alphabet atom space with the non-terminals 
```

Comment:
Would be nice to have a white space between previous function and next function or next function's documentation for better readbility
```suggestion

;; This function initializes the alphabet atom space with the non-terminals 
```